pkg/ccl/backupccl/backup_planning.go

```diff
@@ -1298,6 +1298,18 @@ func backupPlanHook(
 			CreatedBy: backupStmt.CreatedByInfo,
 		}
 
+		spansToProtect := spans
+		// If this is a full cluster backup from the system tenant then we write a
+		// single protected timestamp record spanning the entire keyspace.
+		if backupStmt.Coverage() == tree.AllDescriptors {
+			if p.ExecCfg().Codec.ForSystemTenant() {
+				spansToProtect = []roachpb.Span{{
```

Comment:
We'll also need to protect the system tables we include in the backup, right?

Reply:
I think `TableDataMin` protects everything starting from table ID 0 so it should include the system tables? https://github.com/adityamaru/cockroach/blob/eb63e71b01d889d5e2f3412143dc31fa0552cd4c/pkg/keys/constants.go#L290-L290

Reply:
Ah, indeed -- I thought it was `UserTableDataMin`. I wonder if expanding to include all system tables is more aggressive than we need to be since there are some tables that explicitly lower their GC TTL. Could we maybe merge the spans with (UserTableDataMin, TableDataMax) so we keep the table-level resolution for the system tables and cover the user data wholesale?

Reply:
That being said, maybe protecting them is ~ok given that we should probably stay away from using `UserTableDataMin` since it will probably interact poorly once we extend system tables to exist beyond 50. Perhaps if system tables really should not be included they will be added to a set of descriptors that are explicitly excluded.

Reply:
That's a good point. I guess we don't really guarantee which system tables we will/will not backup as part of a vanilla cluster backup and so we have some wiggle room to say we treat them all the same...I like the idea of having exclude teach the backup to protect less.

Reply:
I'm fine protecting them for the duration of a backup, even if they have short TTLs until we see a strong reason to add exclusions. We're headed to a word of tenant backups where we protect their whole span system tables and all so I'm ambivalent about non-tenant backups being more granular.